synapse/cmds/cron.py

```diff
@@ -0,0 +1,598 @@
+    cron add -H 0,12= -d Wed,Sun= {#foo}
+
+    Run a query every other day at 3:57pm
+    cron add -d 2 -M 57= -H 15 {#foo}
+'''
+
+class Cron(s_cli.Cmd):
+    '''
+Manages cron jobs in a cortex.  Cron jobs are rules persistently stored in a
+cortex such that storm queries automatically run on a time schedule.
+
+Cron jobs maybe be recurring or one-time.  Use the 'at' command to add one-time
+jobs.
+
+A subcommand is required.  Use 'cron -h' for more detailed help.  '''
+    _cmd_name = 'cron'
+
+    _cmd_syntax = (
+        ('line', {'type': 'glob'}),
+    )
+
+    async def _match_idens(self, core, prefix):
+        '''
+        Returns the iden that starts with prefix.  Prints out error and returns None if it doesn't match
+        exactly one.
+        '''
+        idens = [iden for iden, trig in await core.listCronJobs()]
+        matches = [iden for iden in idens if s_common.ehex(iden).startswith(prefix)]
+        if len(matches) == 1:
+            return matches[0]
+        elif len(matches) == 0:
+            self.printf('Error: provided iden does not match any valid authorized cron job')
+        else:
+            self.printf('Error: provided iden matches more than one cron job')
+        return None
+
+    def _make_argparser(self):
+
+        parser = s_cmd.Parser(prog='cron', outp=self, description=self.__doc__)
+
+        subparsers = parser.add_subparsers(title='subcommands', required=True, dest='cmd',
+                                           parser_class=functools.partial(s_cmd.Parser, outp=self))
+
+        subparsers.add_parser('list', help="List cron jobs you're allowed to manipulate", usage=ListHelp)
+
+        parser_del = subparsers.add_parser('del', help='delete a cron job', usage=DelHelp)
+        parser_del.add_argument('prefix', help='Cron jobiden prefix')
+        parser_add = subparsers.add_parser('add', help='add a cron job', usage=AddHelp)
+        parser_add.add_argument('--minute', '-M')
+        parser_add.add_argument('--hour', '-H')
+        parser_add.add_argument('--day', '-d', help='day of week, day of month or number of days')
+        parser_add.add_argument('--month', '-m')
+        parser_add.add_argument('--year', '-y')
+        parser_add.add_argument('query', help='Storm query in curly braces')
+
+        parser_del = subparsers.add_parser('del', help='delete a cron job', usage=DelHelp)
+        parser_del.add_argument('prefix', help='Cron job iden prefix')
+
+        parser_del = subparsers.add_parser('stat', help='details a cron job', usage=StatHelp)
+        parser_del.add_argument('prefix', help='Cron job iden prefix')
+
+        parser_mod = subparsers.add_parser('mod', help='change an existing cron jobquery', usage=ModHelp)
+        parser_mod.add_argument('prefix', help='Cron job iden prefix')
+        parser_mod.add_argument('query', help='New Storm query in curly braces')
+
+        return parser
+
+    @staticmethod
+    def _parse_weekday(val):
+        val = val.title()
```

Comment:
`Try to match a day of the week abbreviation (e.g. Mon), then try to match a full day of the week (e.g. Monday).